arlo_server/all_routes/election/__element__.py

```diff
@@ -0,0 +1,91 @@
+from flask import jsonify, Response, request
+from jsonschema import validate
+
+from arlo_server import app, db
+from arlo_server.auth import require_audit_admin_for_organization
+from arlo_server.models import Election
+
+ELECTION_NAME_KEY = "electionName"
+ELECTION_ONLINE_KEY = "online"
+ELECTION_RANDOM_SEED_KEY = "randomSeed"
+ELECTION_RISK_LIMIT_KEY = "riskLimit"
+ELECTION_STATE_KEY = "state"
+
+PUT_ELECTION_REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        ELECTION_NAME_KEY: {"type": "string"},
+        ELECTION_ONLINE_KEY: {"type": "boolean"},
+        ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+        ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+        ELECTION_STATE_KEY: {"type": "string"},
+    },
+    "additionalProperties": False,
+}
+
+GET_ELECTION_RESPONSE_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "election": {
+            "type": "object",
+            "properties": {
+                ELECTION_NAME_KEY: {"type": "string"},
+                ELECTION_ONLINE_KEY: {"type": "boolean"},
+                ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+                ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+                ELECTION_STATE_KEY: {"type": "string"},
+            },
+            "additionalProperties": False,
+        },
+    },
+    "additionalProperties": False,
+}
+
+
+@app.route("/election/<election_id>", methods=["GET"])
+def get_election(election_id: str) -> Response:
+    election = Election.query.filter_by(id=election_id).one()
+    require_audit_admin_for_organization(election.organization_id)
+
+    response_data = {
+        "election": {
+            ELECTION_NAME_KEY: election.name,
+            ELECTION_ONLINE_KEY: election.online,
+            ELECTION_RANDOM_SEED_KEY: election.random_seed,
+            ELECTION_RISK_LIMIT_KEY: election.risk_limit,
+            ELECTION_STATE_KEY: election.state,
+        }
+    }
+
+    validate(schema=GET_ELECTION_RESPONSE_SCHEMA, instance=response_data)
+
+    return jsonify(response_data)
+
+
+@app.route("/election/<election_id>", methods=["PUT"])
+def put_election(election_id: str) -> Response:
+    election = Election.query.filter_by(id=election_id).one()
+    require_audit_admin_for_organization(election.organization_id)
+
+    settings = request.get_json()
+    validate(schema=PUT_ELECTION_REQUEST_SCHEMA, instance=settings)
+
+    if ELECTION_NAME_KEY in settings:
+        election.name = settings[ELECTION_NAME_KEY]
```

Comment:
I'm realizing that these semantics are for `PATCH`, not `PUT`, which is my mistake. I'm thinking we should either:
- Change to `PATCH`
- Require every field be sent like a regular `PUT`, but accept null values

My leaning is towards the latter for consistency with other APIs. A world in which we always PUT the whole object seems simpler than a world in which some APIs PUT and some PATCH.

What do you think?

Reply:
I'm fine with making it `PUT`.

Reply:
@MorganLove do you have any thoughts on this? Are you storing the election settings in a way that makes it easy for you to send them all to the backend at once with a `PUT` request?

Reply:
@jonahkagan If the information is all on one screen in the setup stage, then it's all able to be sent at once. It's when it's split across multiple screens that things get sketchy. I think all the audit settings data is in that one screen, though, so we should be good.

Reply:
They are all on one screen except `state`, I believe.